packages/workers-shared/asset-server-worker/src/assets-manifest.ts

```diff
@@ -6,6 +12,82 @@ export class AssetsManifest {
 	}
 
 	async get(pathname: string) {
-		return Promise.resolve(pathname);
+		// encode pathname to match manifest encoding
+		const pathHash = await hashPath(pathname);
+		const entry = binarySearch(
+			new Uint8Array(this.data, HEADER_SIZE),
+			pathHash
+		);
+		return entry ? contentHashToKey(entry) : null;
 	}
 }
+
+const hashPath = async (path: string) => {
+	const encoder = new TextEncoder();
+	const data = encoder.encode(path);
+	const hashBuffer = await crypto.subtle.digest("SHA-256", data.buffer);
+	return new Uint8Array(hashBuffer, 0, PATH_HASH_SIZE);
+};
+
+const binarySearch = (
```

Comment:
I always wanted to swap this out for an interpolation search which should be slightly more efficient that binary because our search space is sorted and uniformly distributed, but I never got around to it. Now that it's here and available so easily for testing in `wrangler dev`, it'll be much easier to swap out in the future :)

Reply:
if we have time at the end (lol), happy to add that in for you :)